### PR TITLE
Modify replica struct to better describe deal spanning

### LINKED ISSUE #65: Replica Struct When Blob Spans Multiple Deals
# What

The way we describe replicas seems not super well suited to a case where a blob (>32GB) is spread across multiple pieces.
(see https://github.com/filecoin-project/motion/blob/main/blob/blob.go#L33)

If a blob has multiple pieces, then the deals for each of those pieces will have different statuses, verification times, etc.

I wonder if we need to make the structure:

````
type (
Piece struct {
                 PieceCID []bytes
		Status       string
		LastVerified time.Time
		Expiration   time.Time
}

Replica struct {
		Provider     string
}
)
```
Or something to that effect.

Not super urgent but want to flag.

### PR DESCRIPTION
**Pull request by gammazero**
This PR changes a replica to be composed of a slice of pieces, each with a different status, verification time, etc. as per #65.

Fixes #65

### CODE REVIEW

**Review by masih — APPROVED** [review]
LGTM, we also need to update the HTTP api spec.

File: api/model.go [hunk 1]
```diff
@@ -14,13 +14,17 @@ type (
 		Error string `json:"error"`
 	}
 	GetStatusResponse struct {
-		ID       string    `json:"id"`
-		Replicas []Replica `json:"Replicas,omitempty"`
+		ID      string   `json:"id"`
+		Replica *Replica `json:"Replicas,omitempty"`
```
Comment (masih) [inline api/model.go:18]: 🤔 this should continue to be a slice right? A blob may have multiple replicas.

Comment (gammazero) [reply api/model.go]: That seems correct. I was a bit confused by the relationship between deals and pieces.  I think something needs to change [here](https://github.com/filecoin-project/motion/pull/153/files#diff-da5cbb9a08e25011c2c5c0ea433286ca1b844b015f8be8dca6da93ae7df99025R469-R482)

Does each deal map to a separate replica, or a separate piece, or either? So if a blob is spread across multiple pieces, are those pieces in separate deals? If so, then I need to know how to determine if separate deals from [`getFileDealsRes`](https://github.com/filecoin-project/motion/pull/153/files#diff-da5cbb9a08e25011c2c5c0ea433286ca1b844b015f8be8dca6da93ae7df99025R469) are for different replicas or for different pieces in the same replica.

**Review by masih — APPROVED** [review]

File: api/model.go (hunk 1 above)
Comment (masih) [reply api/model.go]: > something needs to change [here](https://github.com/filecoin-project/motion/pull/153/files#diff-da5cbb9a08e25011c2c5c0ea433286ca1b844b015f8be8dca6da93ae7df99025R469-R482)

Right, the work here is to set up stepping stones for further changes across the workflow.

> Does each deal map to a separate replica, or a separate piece, or either?

Mapping of replica:deal:piece is 1:1:1 for blobs that fit within a sector (32GiB currently). But as soon as blobs become larger, then the mapping of replica to deal looks something like this 1:1..*, and mapping of deal to piece (for the time being) continues to be 1:1.

Cc @xinaxu We should think about large blob support and mapping of Singularity primitives to Motion as part of work items beyond Beta.

**Pull request merged**